modules/core/src/test/java/org/apache/ignite/internal/metric/SystemViewSelfTest.java

```diff
@@ -2044,6 +2048,53 @@ public void testBaselineNodeAttributes() throws Exception {
         }
     }
 
+    /** */
+    @Test
+    public void testSnapshots() throws Exception {
+        cleanPersistenceDir();
+
+        String dfltCacheGrp = "testGroup";
+
+        String testSnap0 = "testSnap0";
+        String testSnap1 = "testSnap1";
+
+        try (IgniteEx ignite = startGrid(getConfiguration()
+            .setCacheConfiguration(new CacheConfiguration<>(DEFAULT_CACHE_NAME).setGroupName(dfltCacheGrp))
+            .setDataStorageConfiguration(
+                new DataStorageConfiguration().setDefaultDataRegionConfiguration(
+                    new DataRegionConfiguration().setName("pds").setPersistenceEnabled(true)
+                )))
+        ) {
+            ignite.cluster().state(ClusterState.ACTIVE);
+
+            ignite.cache(DEFAULT_CACHE_NAME).put(1, 1);
+
+            ignite.snapshot().createSnapshot(testSnap0).get();
+
+            SystemView<SnapshotView> views = ignite.context().systemView().view(SNAPSHOTS_SYS_VIEW);
+
+            assertEquals(1, F.size(views.iterator()));
+
+            SnapshotView view = views.iterator().next();
+
+            assertEquals(testSnap0, view.snapshotName());
+            assertEquals(toStringSafe(ignite.localNode().consistentId()), view.nodeId());
+
+            List<?> collect = ignite.cluster().nodes().stream().map(ClusterNode::consistentId).collect(Collectors.toList());
+
+            assertEquals(toStringSafe(collect), view.baselineNodes());
+            assertEquals(toStringSafe(Arrays.asList(dfltCacheGrp, METASTORAGE_CACHE_NAME)), view.cacheGroups());
+
+            ignite.createCache("testCache");
+
+            ignite.snapshot().createSnapshot(testSnap1).get();
+
+            views = ignite.context().systemView().view(SNAPSHOTS_SYS_VIEW);
+
+            assertEquals(2, F.size(views.iterator()));
```

Comment:
Can we use `views.size()` instead of `F.size(views.iterator()`?

Reply:
fixed